pkg/services/rpcsrv/subscription_test.go

```diff
@@ -582,25 +583,38 @@ func TestWSClientsLimit(t *testing.T) {
 			dialer := websocket.Dialer{HandshakeTimeout: 5 * time.Second}
 			url := "ws" + strings.TrimPrefix(httpSrv.URL, "http") + "/ws"
 			wss := make([]*websocket.Conn, effectiveClients)
+			var wg sync.WaitGroup
 
-			for i := 0; i < len(wss)+1; i++ {
-				ws, r, err := dialer.Dial(url, nil)
-				if r != nil && r.Body != nil {
-					defer r.Body.Close()
-				}
-				if i < effectiveClients {
+			// Dial effectiveClients connections in parallel
+			for i := 0; i < effectiveClients; i++ {
+				wg.Add(1)
+
+				go func(i int) {
+					defer wg.Done()
+					ws, r, err := dialer.Dial(url, nil)
+					if r != nil {
+						defer r.Body.Close()
+					}
 					require.NoError(t, err)
 					wss[i] = ws
-					// Check that it's completely ready.
 					doSomeWSRequest(t, ws)
-				} else {
-					require.Error(t, err)
-				}
+				}(i)
```

Comment:
Oh-oh. https://go.dev/blog/loopvar-preview

Reply:
I've tried it locally, and actually it will work properly when you pass `i` as an argument. But for direct loopvar usage in function - it won't.